examples/2_simple_with_filter/main.go

```diff
@@ -14,23 +14,24 @@ import (
 )
 
 func main() {
-	logger := logger.NewLogrus(false)
+	// Instantiate a logger plugin
+	log := logger.NewLogrus(false)
 
-	inventory, err := inventory.FromYAMLFile("/go/src/github.com/nornir-automation/gornir/examples/hosts.yaml")
+	// Load the inventory using the FromYAMLFile plugin
+	file := "/go/src/github.com/nornir-automation/gornir/examples/hosts.yaml"
+	plugin := inventory.FromYAML{HostsFile: file}
+	inv, err := plugin.Create()
 	if err != nil {
-		logger.Fatal(err)
-	}
-
-	gr := &gornir.Gornir{
-		Inventory: inventory,
-		Logger:    logger,
+		log.Fatal(err)
 	}
 
 	// define a function we will use to filter the hosts
 	filter := func(ctx context.Context, h *gornir.Host) bool {
 		return h.Hostname == "dev1.group_1" || h.Hostname == "dev4.group_2"
 	}
 
+	gr := gornir.New().WithInventory(inv).WithLogger(log).WithFilter(filter)
```

Comment:
`WithFilter(filter)` is redundant here as it's called a couple of lines below.

Reply:
I'd prefer to keep it there, and remove it in the other call, so we end up with:

```go
gr := gornir.New().WithInventory(inv).WithLogger(log).WithFilter(filter)

results, err := gr.RunSync(
		"What's my ip?",
		runner.Parallel(),
		&task.RemoteCommand{Command: "ip addr | grep \\/24 | awk '{ print $2 }'"},
	)
```

Which in my view, separate the two actions (building the inventory and running the task). But this is just my personal preference.

Reply:
I want to keep the old pattern to illustrate the builder pattern, people is free to use whatever pattern they prefer in their code but I want to make sure people sees something like this is possible:

```
gr := gornir.New()
...
results, err := gr.Filter(giveMeSpines).Run(...)
...
results, err = gr.Filter(giveMeLeaves).Run(...)
```

Reply:
Ok, I'll keep that example with this alternative and let the user choose which one they prefer in their code. Thanks.